gqlalchemy/transformations/translators/nx_translator.py

```diff
@@ -0,0 +1,279 @@
+        properties_without_labels = {k: v for k, v in properties.items() if k != NetworkXGraphConstants.LABELS}
+        properties_str = to_cypher_properties(properties_without_labels, self._config)
+
+        return f"CREATE ({labels_str} {properties_str});"
+
+    def _create_edge(
+        self,
+        from_id: Union[int, str],
+        to_id: Union[int, str],
+        from_label: Union[str, List[str]],
+        to_label: Union[str, List[str]],
+        properties: Dict[str, Any],
+    ) -> str:
+        """Returns Cypher query for edge creation."""
+        edge_type = to_cypher_labels(properties.get(NetworkXGraphConstants.TYPE, "TO"))
+        properties.pop(NetworkXGraphConstants.TYPE, None)
+        properties_str = to_cypher_properties(properties, self._config)
+        from_label_str = to_cypher_labels(from_label)
+        to_label_str = to_cypher_labels(to_label)
+        from_id_str = to_cypher_value(from_id)
+        to_id_str = to_cypher_value(to_id)
+
+        return f"MATCH (n{from_label_str} {{id: {from_id_str}}}), (m{to_label_str} {{id: {to_id_str}}}) CREATE (n)-[{edge_type} {properties_str}]->(m);"
+
+    def _create_index(self, label: str, property: str = None):
+        """Returns Cypher query for index creation."""
+        index = MemgraphIndex(label, property)
+        return f"CREATE INDEX ON {index.to_cypher()}(id);"
+
+
+class NoNetworkXConfigException(Exception):
+    pass
+
+
+class NxTranslator(Translator):
+    """Uses original ids from Memgraph. Labels are encoded as properties. Since Networkx allows
+    that nodes have properties of different dimensionality, this modules makes use of it and stores properties
+    as dictionary entries. All properties are saved to Networkx data structure.
+    """
+
+    def __init__(self, default_node_label="NODE", default_edge_type="RELATIONSHIP",
+        host: str = MG_HOST,
+        port: int = MG_PORT,
+        username: str = MG_USERNAME,
+        password: str = MG_PASSWORD,
+        encrypted: bool = MG_ENCRYPTED,
+        client_name: str = MG_CLIENT_NAME,
+        lazy: bool = MG_LAZY,) -> None:
+            self.__all__ = ("nx_to_cypher", "nx_graph_to_memgraph_parallel")
+            super().__init__(default_node_label, default_edge_type, host, port, username, password, encrypted, client_name, lazy)
+
+
+    def to_cypher_queries(self, graph: nx.Graph, config: NetworkXCypherConfig = None) -> Iterator[str]:
+        """Generates a Cypher query for creating a graph."""
+
+        if config is None:
+            config = NetworkXCypherConfig()
+
+        builder = NetworkXCypherBuilder(config=config)
+
+        yield from builder.yield_queries(graph)
+
+    def nx_graph_to_memgraph_parallel(
+        self,
+        graph: nx.Graph,
+        host: str = "127.0.0.1",
+        port: int = 7687,
+        username: str = "",
+        password: str = "",
+        encrypted: bool = False,
```

Comment:
Why is `host`, `port`, `username`, `password` mentioned here again? Because we have it in init, right?

Reply:
Yes, this method was from old times so I forgot to change this.